bin/main.rb

```diff
@@ -1,3 +1,84 @@
 #!/usr/bin/env ruby
 
-puts 'Hello World!'
+def user_input(player)
+  user_choice = false
+
+  puts "please #{player} choose the square that you want to put your mark in:"
+  loop do
+    user_choice = gets.chomp.to_i
+    if user_choice < 10 && user_choice.positive?
+      # treat player choice
+      puts ''
+      break
+    else
+      puts 'You should select a number between 1 and 9'
+    end
+  end
+
+  user_choice
+end
+
+def play_again
+  answer = false
+
+  loop do
+    puts 'do you want to play again?(Y/N)'
+    answer = gets.chomp
+    arr = %w[Y y N n]
+    break if arr.include?(answer)
+
+    puts 'Wrong input you must choose Y or N'
+  end
+
+  answer
+end
+
+def game(player1, player2)
+  9.times do |number|
```

Comment:
Instead of looping an x amount of time, it'd be better to have have an on or off switching indicating if the game is on or off. For example, we could do something like:

```
game_on = true

# initial instructions for player

while game_on
  # loop for each move
  
  if winner # and/or draw (the exact condition in this milestone is not important)
    game_on = false
  end
end
```

I understand that this is done this way for now, and as such, I'll let this one pass, but please consider this suggestion [`OPTIONAL`].